client/src/lib/appState/hooks/useToggleHiddenContent.ts

```diff
@@ -0,0 +1,24 @@
+import {useCallback} from 'react';
+import {useTranslation} from 'react-i18next';
+import {LANGUAGE_TAGS} from '../../i18n';
+import useAppState from '../state/state';
+
+const useToggleHiddenContent = () => {
+  const {i18n} = useTranslation();
+  const setShowHiddenContent = useAppState(state => state.setShowHiddenContent);
+
+  const toggleHiddenContent = useCallback(
+    (on: boolean) => {
+      setShowHiddenContent(on);
+      // Needs to be removed for all languages,
+      // otherwise they are not reloaded when switching language
+      LANGUAGE_TAGS.forEach(lng => i18n.removeResourceBundle(lng, 'exercises'));
+      i18n.reloadResources(undefined, 'exercises');
+    },
+    [i18n, setShowHiddenContent],
+  );
+
+  return toggleHiddenContent;
+};
+
+export default useToggleHiddenContent;
```

Comment:
Could we move this to `i18n` lib instead?

Reply:
Right, forgot in my frustration, will do